platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -344,13 +344,30 @@ Previously, extensions registered in `Plugin<Settings>` weren't available in `se
 Now, type-safe accessors for these extensions are generated.
 
 This fixes https://github.com/gradle/gradle/issues/11210.
-<!-- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-ADD RELEASE FEATURES ABOVE
-==========================================================
 
--->
+#### Improved control of local build cache cleanup
+
+Previously, cleanup of the local build cache (`~/.gradle/caches/build-cache-1`) ran every 24 hours, and this interval could not be configured.
+The retention period was configured via the `DirectoryBuildCache.removeUnusedEntriesAfterDays` setting. 
+
+With this release, local build cache cleanup is configurable via the [standard init-script mechanism](userguide/directory_layout.html#dir:gradle_user_home:cache_cleanup),
+providing improved control and consistency.
+- Setting `Settings.caches.cleanup = Cleanup.DISABLED` or `Settings.caches.cleanup = Cleanup.ALWAYS` will now prevent or force the cleanup of the local build cache
+- Build cache entry retention is now configured via `Settings.caches.buildCache.setRemoveUnusedEntriesAfterDays()`
+
+To configure cleanup for the local build cache, add an `init-script` to `~/.gradle/init.d`:
+```groovy
+beforeSettings {
+    caches {
+        cleanup = Cleanup.ALWAYS  // Clean Gradle User Home caches after every build
+        buildCache.setRemoveUnusedEntriesAfterDays(5) // Retain local build-cache entries for 5 days
+    }
+}
+```
+
+The `DirectoryBuildCache.removeUnusedEntriesAfterDays` method is now deprecated.
```

Comment:
🤔 Should we mention that if `DirectoryBuildCache.removeUnusedEntriesAfterDays` is set, it will take precedence for backward compatibility?

Reply:
Good point. I've added a statement to that effect.

Reply:
I also significantly improved the upgrade guide section based on the release notes.
I'm going to merge this in now. We can always improve it later if necessary.